dashboard/app/components/ui/sidebar/app.sidebar.tsx

```diff
@@ -0,0 +1,184 @@
+'use client'
+import * as React from "react"
+import { ChevronRight, ChevronLeft, FileText, LineChart, Binary } from 'lucide-react'
+
+import { 
+  Sidebar,
+  SidebarContent,
+  SidebarHeader,
+  SidebarMenu,
+  SidebarMenuButton,
+  SidebarMenuItem,
+  SidebarMenuSub,
+  SidebarMenuSubButton,
+  SidebarMenuSubItem,
+  SidebarRail,
+  SidebarTrigger,
+  SidebarFooter,
+  useSidebar,
+} from "~/components/ui/sidebar"
+import { cn } from "~/utils/common"
+import { useLocation } from "react-router"
```

Comment:
I think this is wrong

Reply:
Yup, using useLocation from react-router-dom requires its parent component to be wrapped in a <Router> component because it needs Router's context to work.